test-integration/graql/reasoner/query/AtomicQueryUnificationIT.java

```diff
@@ -555,11 +555,39 @@ public void testUnification_RelationsWithVariableRolesAndPotentialTypes() {
             unification(query, potentialEquivalent, true, UnifierType.RULE, testTx);
             unification(potentialEquivalent, query, true, UnifierType.RULE, testTx);
 
-            unification(query, potentialEquivalent, true, UnifierType.SUBSUMPTIVE, testTx);
+            // we block subtype checking by tying the type of $role in the potentialEquivalent
+            unification(query, potentialEquivalent, false, UnifierType.SUBSUMPTIVE, testTx);
```

Comment:
I don't follow how blocking subtype checking matters - isn't the unification subsumptive if the two would share the exact same set of answers? Which, in this case, it seems they will. I must be missing something.

Reply:
Key point here is that `query` doesn't have answers that are a subtype of `potentialQuery`, but only the other way around. 
This is because of key point 2: by specifying the type of `$role` ie `$role type baseRole1`, we are preventing the inheritance mechanism from kicking in (exemplifies the edge cases i was talking about reasoner has to handle!)

So, to recap, all the answers of `query` include the answers to `potentialEquivalent`. However, the reverse is not true (see line below for other direction)

Reply:
Aha - in `query` `baseRole1` can be itself or any of its subtypes, whereas in `potentialEquivalent`, `$role1` is constrained to be exactly `baseRole1` and not any of its subtypes. The new test also makes sense given this.

Reply:
Ok so this one is actually false because of a different reason - the second query has an extra variable, so it can never be subsumed by the one with fewer variables :)

Reply:
I just had the thought - if this is false because of the extra variable, do we have a test case which models the behaviour when each has the same variables? I feel that would be useful to capture the full context of what's going on here.

Reply:
yep these exists already im quite sure!